src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -31,0 +1,68 @@
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  allTodos: Todo[];
+  setTodos: (todos: Todo[]) => void;
+};
+
+function getActiveTodos(todos: Todo[]): Todo[] {
+  return todos.filter(todo => todo.completed === false);
+}
+
+function getCompletedTodos(todos: Todo[]): Todo[] {
+  return todos.filter(todo => todo.completed === true);
+}
+
+export const TodoFilter: React.FC<Props> = ({ allTodos, setTodos }) => {
+  const activeTodos = getActiveTodos(allTodos);
+  const completedTodos = getCompletedTodos(allTodos);
+
+  const handleTodoChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    switch (event.target.value) {
+      case 'all':
+        setTodos(allTodos);
+        break;
+      case 'active':
+        setTodos(activeTodos);
+        break;
+      case 'completed':
+        setTodos(completedTodos);
+        break;
+    }
+  };
+
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select data-cy="statusSelect" onChange={handleTodoChange}>
+            <option value="all">All</option>
+            <option value="active">Active</option>
+            <option value="completed">Completed</option>
+          </select>
+        </span>
+      </p>
+
+      <p className="control is-expanded has-icons-left has-icons-right">
+        <input
+          data-cy="searchInput"
+          type="text"
+          className="input"
+          placeholder="Search..."
+        />
+        <span className="icon is-left">
+          <i className="fas fa-magnifying-glass" />
+        </span>
+
+        <span className="icon is-right" style={{ pointerEvents: 'all' }}>
+          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+          <button
+            data-cy="clearSearchButton"
+            type="button"
+            className="delete"
+          />
```

Comment:
The clear search button is present but not functional. Implement an event handler to clear the search input and reset the todos list when this button is clicked.